test/mocha/e2e/quote.test.ts

```diff
@@ -2725,9 +3044,49 @@ describe('quote', function () {
             const response: AxiosResponse<QuoteResponse> = await axios.get<QuoteResponse>(`${API}?${queryParams}`, {
               headers: headers,
             })
-            const { status } = response
+            const { status, data } = response
 
             expect(status).to.equal(200, JSON.stringify(response.data))
+
+            let nativeOrWrappedNativePoolFound = false
+            let nativePoolFound = true
+
+            // the v4 native pools or v3 wrapped native pools on sepolia can grow exponentially,
+            // we just have to iterate through to make sure find it and assert the important data
+            data.route.forEach((pools) => {
+              pools.forEach((pool) => {
+                if (
+                  chain === ChainId.SEPOLIA &&
+                  pool.tokenIn.address === '0x0000000000000000000000000000000000000000'
+                ) {
+                  nativeOrWrappedNativePoolFound = true
+                  nativePoolFound = true
+                }
+
+                if (
+                  chain !== ChainId.SEPOLIA &&
+                  pool.tokenIn.address.toLowerCase() === WNATIVE_ON(chain).address.toLowerCase()
+                ) {
+                  nativeOrWrappedNativePoolFound = true
+                }
+              })
+            })
+
+            if (chain === ChainId.SEPOLIA) {
+              expect(nativePoolFound).to.be.true
```

Comment:
this should be true not for sepolia but instead for v4 only right?

Reply:
yeah definitely, i think this is correct now but soon will be stale, so we gotta keep it real

Reply:
then again, what if these pools are not selected because of liqudiity even though they are there, what would we do?

Reply:
the test is written that it either requests v4, or requests v2 & v3, so that it must route through v4, as long as v4 is supported on this chain.

Reply:
the only case then becomes if entire ETH/USDC v4 pool liquidity is withdrawn from the LPers, which is unlikely